app/models/sale_item.rb

```diff
@@ -116,6 +118,12 @@ class SaleItem < Ekylibre::Record::Base
     joins(:variant).merge(ProductNatureVariant.of_natures(product_nature))
   }
 
+  scope :active, -> { includes(:sale).where.not(sales: { state: %i[refused aborted] }).order(created_at: :desc) }
+  scope :invoiced_on_or_after, -> (date) { includes(:sale).where("invoiced_at >= '#{date}' OR invoiced_at IS NULL") }
```

Comment:
`where("invoiced_at >= ? OR invoiced_at IS NULL", date)` would be better. I suppose there is also a `or` method in the builder to avoid having to write SQL chunks.